core/client.go

```diff
@@ -152,6 +157,22 @@ func (c *client) IsSubscribed(channel string) bool {
 	return false
 }
 
+func (c *client) Unsubscribe(message *messages.Unsubscribe) {
+	if debug {
+		log.Printf("Client from %s unsubscribing from %s", c.GetRemoteAddr(), message.Channel)
+	}
+
+	// Return
+	if !c.IsSubscribed(message.Channel) {
+		c.SendMessage(messages.NewGenericMessage(messages.TYPE_UNSUBSCRIBABLE_CHANNEL))
+		c.Close()
+		return
+	}
+
+	unsubscribe(message.Channel, c)
+	c.SendMessage(messages.NewGenericMessage(messages.TYPE_UNSUBSCRIBE_OK))
```

Comment:
We don't need the `unsubscribe_ok` event, there is no need to make the clients wait for that.